packages/lesswrong/components/ultraFeed/UltraFeedThreadItem.tsx

```diff
@@ -237,6 +261,84 @@ const UltraFeedThreadItem = ({thread, index, settings = DEFAULT_SETTINGS}: {
     }
   }, [comments, commentDisplayStatuses, setDisplayStatus, captureEvent]);
 
+  const handleReplyClick = (commentId: string) => {
+    setReplyingToCommentId(commentId);
+  };
+
+  const handleReplySubmit = (parentCommentId: string, newComment: UltraFeedComment) => {
+    setNewReplies(prev => ({
+      ...prev,
+      [parentCommentId]: newComment
+    }));
+    setBranchViewStates(prev => ({
+      ...prev,
+      [parentCommentId]: 'new'
+    }));
+    setReplyingToCommentId(null);
+    
+    const defaultMetaInfo: FeedCommentMetaInfo = {
+      displayStatus: 'expanded',
+      sources: [],
+      directDescendentCount: 0,
+      highlight: false,
+      lastServed: new Date(),
+      lastViewed: null,
+      lastInteracted: new Date(),
+      postedAt: newComment.postedAt ? new Date(newComment.postedAt) : new Date(), //TODO: this is coming back as string from new comment creation, is that right?
```

Comment:
yes, we've always been getting dates as strings on the client and we just had incorrect types for a long time

Reply:
(you can use the `maybeDate` helper function here instead: `postedAt: maybeDate(newComment.postedAt) ?? new Date()`)